moh-kenya/echis-2.0-integration
# Added Multi-Tenant Capabilities for Different eCHIS Instances. [uses X-OpenHIM-ClientID in headers]

**Description**
Link to Github Ticket: https://github.com/moh-kenya/echis-2.0-integration/issues/121

**What does this PR do?:**
This Pull request introduces the use of X-OpenHIM-ClientID headers coming from OpenHIM as here: 

<img width="680" alt="image" src="https://github.com/moh-kenya/echis-2.0-integration/assets/18611345/aabcc80d-56de-40ec-848d-4e1379407e65">

This header is passed when a client is setup and the same client is used in echis as seen here:

<img width="680" alt="image" src="https://github.com/moh-kenya/echis-2.0-integration/assets/18611345/d74c0a08-b51c-4b89-a680-a30c97d6f0ba">

**How was the solution implemented?**
1. Adding an .env file that contains all credentials for the various instances.
2. Created a function in the mediator to handle the request header and sets the instance in a global variable.
3. Changed CHT object to a function to allow it to pick values dynamically from env.
4. Route to the right eCHIS instance based on the picked values from env.

**How can the changes be tested?**
The changes can be deployed to staging to test. By default echis has been handled. This can be setup locally for both echis and mediator. 

Ensure the env file is updated accordingly.
